src/components/Header.jsx

```diff
@@ -3,38 +3,61 @@ import logoImg from "../assets/logo@2x.png";
 import logoText from "../assets/logo_text@2x.png";
 import "../index.css";
 import { useLocation } from "react-router-dom";
+import KaKaoLogo from "../assets/kakao_logo@3x.png";
+import useMediaQuery from "../utils/useQueryMedia";
+
+function HeaderLogo() {
+  const isNotMobile = useMediaQuery("(min-width: 767px)");
+
+  return (
+    <>
+      {isNotMobile ? (
+        <img
+          className="header__desktop-img"
+          src={logoImg}
+          alt="logo"
+          width="153"
+          height="51"
+        />
+      ) : (
+        <img
+          className="header__mobile-img"
+          src={logoText}
+          alt="logo"
+          width="103"
+          height="35"
+        />
+      )}
+    </>
+  );
```

Comment:
```suggestion
const logoData = {
  desktop: {
    className: "header__desktop-img",
    src: logoImg,
    width: 153,
    height: 51,
    alt: "desktop logo",
  },
  mobile: {
    className: "header__mobile-img",
    src: logoText,
    width: 103,
    height: 35,
    alt: "mobile logo",
  },
};

function HeaderLogo() {
  const isNotMobile = useMediaQuery({ query: "(min-width: 767px)" });

  const logo = isNotMobile ? logoData.desktop : logoData.mobile;

  return (
    <img
      className={logo.className}
      src={logo.src}
      alt={logo.alt}
      width={logo.width}
      height={logo.height}
    />
  );
}
```

참고로 이렇게 데이터랑 render되는 JSX랑 분리해서 관리할 수도 있습니다. 
이 방식은 서버에서 데이를 받아 UI로 구현하는 방식이랑 비슷합니다.